### DISCUSSION TITLE
Does wezterm also have this feature? I can't find the documentation

### DISCUSSION THREAD

**Discussion by skoowoo** [wezterm/wezterm#3839] [Q&A]
Usually I like the padding for window, but I want to disable padding while using vim.

https://www.reddit.com/r/vim/comments/ofe72k/disabling_terminal_padding_while_using_vim/

**Comment by nekowinston** [comment]
I have something like this in [my config](https://github.com/nekowinston/dotfiles/blob/main/home/apps/wezterm/smartpadding.lua)
```lua
wezterm.on("update-status", function(window, pane)
  if pane:is_alt_screen_active() then
    window:set_config_overrides({
      window_padding = { left = 0, right = 0, top = 0, bottom = 0 },
    })
  else
    window:set_config_overrides({
      window_padding = { left = 5, right = 5, top = 5, bottom = 5 },
    })
  end
end)
```
This also works for `less`, `bat`, etc.

Alternatively, you could try to match against `pane:get_foreground_process_info().name` to only handle vim:
```lua
wezterm.on("update-status", function(window, pane)
  local info = pane:get_foreground_process_info()
  if info == nil then
    return
  end
  if info.name == "nvim" then
    window:set_config_overrides({
      window_padding = { left = 0, right = 0, top = 0, bottom = 0 },
    })
  else
    window:set_config_overrides({
      window_padding = { left = 5, right = 5, top = 5, bottom = 5 },
    })
  end
end)
```

**Reply by beeb** [reply]
Is there a way to apply the config overrides to one pane only using the second method (looking at the process name)?

EDIT: i saw your reply below. Not sure if anything has changed since July.

**Reply by nekowinston** [reply]
hasn't changed since then AFAIK.

**Comment by skoowoo** [comment]
thanks for your config.  It's great. hah

**Comment by ryanmsnyder** [comment]
Thanks for the config. It does work but it causes some movement if you have multiple panes open. For example, if you're focused on a NeoVim pane and then navigate to a non-NeoVim pane, the whole window is re-rendered causing both panes to move a little bit. Is there anyway around this? I assume not but wanted to check.

**Reply by nekowinston** [reply]
Yeah, the config overrides are being applied to the [`Window`](https://wezfurlong.org/wezterm/config/lua/window/index.html) object, there's currently no way to apply config on a [`Pane`](https://wezfurlong.org/wezterm/config/lua/pane/index.html).
